## Causas de Morte ao redor do mundo

### Análise Exploratória

*Importanto Base de dados e imprimindo as primeiras 5 linhas*

In [92]:
import pandas as pd

# Importando base de dados 
df = pd.read_csv("./cause_of_deaths.csv")

# Primeiras 5 linhas do Dataframe
df.head()

Country/Territory Code  Year  Meningitis  \
0       Afghanistan  AFG  1990        2159   
1       Afghanistan  AFG  1991        2218   
2       Afghanistan  AFG  1992        2475   
3       Afghanistan  AFG  1993        2812   
4       Afghanistan  AFG  1994        3027   

   Alzheimer's Disease and Other Dementias  Parkinson's Disease  \
0                                     1116                  371   
1                                     1136                  374   
2                                     1162                  378   
3                                     1187                  384   
4                                     1211                  391   

   Nutritional Deficiencies  Malaria  Drowning  Interpersonal Violence  ...  \
0                      2087       93      1370                    1538  ...   
1                      2153      189      1391                    2001  ...   
2                      2441      239      1514                    2299  ...   
3                      2837      108      1687                    2589  ...   
4                      3081      211      1809                    2849  ...   

   Diabetes Mellitus  Chronic Kidney Disease  Poisonings  \
0               2108                    3709         338   
1               2120                    3724         351   
2               2153                    3776         386   
3               2195                    3862         425   
4               2231                    3932         451   

   Protein-Energy Malnutrition  Road Injuries  Chronic Respiratory Diseases  \
0                         2054           4154                          5945   
1                         2119           4472                          6050   
2                         2404           5106                          6223   
3                         2797           5681                          6445   
4                         3038           6001                          6664   

   Cirrhosis and Other Chronic Liver Diseases  Digestive Diseases  \
0                                        2673                5005   
1                                        2728                5120   
2                                        2830                5335   
3                                        2943                5568   
4                                        3027                5739   

   Fire, Heat, and Hot Substances  Acute Hepatitis  
0                             323             2985  
1                             332             3092  
2                             360             3325  
3                             396             3601  
4                             420             3816  

[5 rows x 34 columns]

*Quantas entradas temos? de quantos anos? de quantos países? e de quantas doenças?*

In [93]:
df.shape

(6120, 34)

In [94]:
df['Year'].nunique()

30

In [95]:

df['Country/Territory'].nunique()

204

In [96]:
dfCauses = df.drop(columns=["Country/Territory", "Year"]).reset_index(drop=True)
dfCauses = dfCauses.melt(id_vars=["Code"], var_name="Cause", value_name="Deaths")
dfCauses["Cause"].nunique()

31

Nota-se que há 6120 entradas, recorrentes a informações de 204 países ao longo de 30 anos a respeito de 31 causas de mortes distintas.

Abaixo estão listadas as causas de mortes que estamos trabalhando:

In [97]:
dfCauses["Cause"].unique()

array(['Meningitis', "Alzheimer's Disease and Other Dementias",
       "Parkinson's Disease", 'Nutritional Deficiencies', 'Malaria',
       'Drowning', 'Interpersonal Violence', 'Maternal Disorders',
       'HIV/AIDS', 'Drug Use Disorders', 'Tuberculosis',
       'Cardiovascular Diseases', 'Lower Respiratory Infections',
       'Neonatal Disorders', 'Alcohol Use Disorders', 'Self-harm',
       'Exposure to Forces of Nature', 'Diarrheal Diseases',
       'Environmental Heat and Cold Exposure', 'Neoplasms',
       'Conflict and Terrorism', 'Diabetes Mellitus',
       'Chronic Kidney Disease', 'Poisonings',
       'Protein-Energy Malnutrition', 'Road Injuries',
       'Chronic Respiratory Diseases',
       'Cirrhosis and Other Chronic Liver Diseases', 'Digestive Diseases',
       'Fire, Heat, and Hot Substances', 'Acute Hepatitis'], dtype=object)

Abaixo está disposta a listagem dos anos. Nota-se que trabalharemos no período de tempo 1990 - 2019.

In [98]:
df['Year'].unique()

array([1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000,
       2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011,
       2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019], dtype=int64)

Há dados de todos os países para cada um dos anos?

In [99]:
# Número de entradas (uma por ano) para cada país

df['Country/Territory'].value_counts()


Afghanistan         30
Papua New Guinea    30
Niue                30
North Korea         30
North Macedonia     30
                    ..
Greenland           30
Grenada             30
Guam                30
Guatemala           30
Zimbabwe            30
Name: Country/Territory, Length: 204, dtype: int64

In [100]:
# Verificar se há dados faltantes
df.isnull().sum()

Country/Territory                             0
Code                                          0
Year                                          0
Meningitis                                    0
Alzheimer's Disease and Other Dementias       0
Parkinson's Disease                           0
Nutritional Deficiencies                      0
Malaria                                       0
Drowning                                      0
Interpersonal Violence                        0
Maternal Disorders                            0
HIV/AIDS                                      0
Drug Use Disorders                            0
Tuberculosis                                  0
Cardiovascular Diseases                       0
Lower Respiratory Infections                  0
Neonatal Disorders                            0
Alcohol Use Disorders                         0
Self-harm                                     0
Exposure to Forces of Nature                  0
Diarrheal Diseases                      

Observa-se que nossa base de dados está bem completa, sem dados faltantes. 

Agora, fica a dúvida, qual a principal causa de morte do planeta? 

In [101]:
# Novo dataset com a frequência de cada causa de morte no mundo
dfFrequence = df.groupby("Year").sum(numeric_only=True)
dfFrequence

Meningitis  Alzheimer's Disease and Other Dementias  \
Year                                                        
1990      432253                                   560616   
1991      428621                                   583166   
1992      426440                                   605894   
1993      420836                                   629571   
1994      413799                                   652176   
1995      409826                                   674815   
1996      417259                                   696665   
1997      400893                                   717342   
1998      393364                                   738768   
1999      390136                                   761620   
2000      385994                                   786615   
2001      379912                                   814526   
2002      372298                                   845695   
2003      369171                                   877011   
2004      363462                                   909148   
2005      361809                                   945619   
2006      358973                                   982308   
2007      352747                                  1022057   
2008      343452                                  1065297   
2009      336868                                  1109405   
2010      326778                                  1155944   
2011      312733                                  1201138   
2012      300091                                  1247515   
2013      289502                                  1294701   
2014      280438                                  1343756   
2015      267789                                  1394942   
2016      261246                                  1451840   
2017      250132                                  1509646   
2018      241666                                  1568617   
2019      236084                                  1622426   

      Parkinson's Disease  Nutritional Deficiencies  Malaria  Drowning  \
Year                                                                     
1990               147156                    756808   840297    460460   
1991               150875                    729145   858984    454375   
1992               154886                    700664   856415    447056   
1993               160249                    674219   862216    445434   
1994               164381                    649801   855671    443350   
1995               168882                    723095   862626    437303   
1996               173822                    671977   872476    423296   
1997               179347                    647682   892946    413405   
1998               185097                    620498   901338    407205   
1999               191538                    593417   893788    397169   
2000               198545                    568454   882055    387627   
2001               206194                    542565   912709    375917   
2002               214930                    517754   929195    361967   
2003               222256                    417264   959654    347745   
2004               227695                    393952   960784    338417   
2005               236498                    376740   940540    332705   
2006               242749                    362476   941746    319875   
2007               250475                    347407   940193    311889   
2008               260382                    335623   932910    304870   
2009               267738                    322879   917796    296871   
2010               276188                    322187   909814    290831   
2011               284488                    314327   868564    281106   
2012               293123                    298117   806183    275585   
2013               303488                    291451   751776    266827   
2014               312928                    284578   724324    258489   
2015               321782                    278793   7

In [102]:
# Causa de morte mais frequente no planeta em cada ano
dfMax = dfFrequence.idxmax(axis=1)
dfMax

Year
1990    Cardiovascular Diseases
1991    Cardiovascular Diseases
1992    Cardiovascular Diseases
1993    Cardiovascular Diseases
1994    Cardiovascular Diseases
1995    Cardiovascular Diseases
1996    Cardiovascular Diseases
1997    Cardiovascular Diseases
1998    Cardiovascular Diseases
1999    Cardiovascular Diseases
2000    Cardiovascular Diseases
2001    Cardiovascular Diseases
2002    Cardiovascular Diseases
2003    Cardiovascular Diseases
2004    Cardiovascular Diseases
2005    Cardiovascular Diseases
2006    Cardiovascular Diseases
2007    Cardiovascular Diseases
2008    Cardiovascular Diseases
2009    Cardiovascular Diseases
2010    Cardiovascular Diseases
2011    Cardiovascular Diseases
2012    Cardiovascular Diseases
2013    Cardiovascular Diseases
2014    Cardiovascular Diseases
2015    Cardiovascular Diseases
2016    Cardiovascular Diseases
2017    Cardiovascular Diseases
2018    Cardiovascular Diseases
2019    Cardiovascular Diseases
dtype: object

In [103]:
mode = dfMax.mode()
cases = dfFrequence["Cardiovascular Diseases"].sum(axis=0)
print("A maior causa de morte no planeta de 1990 a 2019:", mode[0], " - tirou a vida de", cases, "individuos.")

A maior causa de morte no planeta de 1990 a 2019: Cardiovascular Diseases  - tirou a vida de 447741982 individuos.


In [104]:
# Outra análise interessante: Qual a causa de morte mais frequente em cada país?
# Para isso, vamos criar um novo dataset com a frequência de cada causa de morte em cada país, em todos os anos

dfCountry = df.drop(columns=["Year"]).reset_index(drop=True)
dfCountry = dfCountry.groupby("Country/Territory").sum(numeric_only=True)
dfCountry


Meningitis  Alzheimer's Disease and Other Dementias  \
Country/Territory                                                        
Afghanistan             78666                                    41998   
Albania                  1323                                    16549   
Algeria                 15685                                    86914   
American Samoa             30                                      143   
Andorra                     0                                      614   
...                       ...                                      ...   
Venezuela               11615                                   108735   
Vietnam                 38559                                   369363   
Yemen                   21095                                    31045   
Zambia                  98886                                    13473   
Zimbabwe                41238                                    20017   

                   Parkinson's Disease  Nutritional Deficiencies  Malaria  \
Country/Territory                                                           
Afghanistan                      13397                     71453    13924   
Albania                           4491                       569        0   
Algeria                          22943                      7138       70   
American Samoa                      69                        60        0   
Andorra                            137                         0        0   
...                                ...                       ...      ...   
Venezuela                        18573                     22554     3726   
Vietnam                          83322                     48613    17311   
Yemen                             7188                     68939   143463   
Zambia                            4054                     95913   205529   
Zimbabwe                          5764                     66723   118728   

                   Drowning  Interpersonal Violence  Maternal Disorders  \
Country/Territory                                                         
Afghanistan           56536                  108228              129621   
Albania                2397                    5242                 246   
Algeria               24273                   16702               29475   
American Samoa          120                     101                  30   
Andorra                   0                      15                   0   
...                     ...                     ...                 ...   
Venezuela             20273                  266071               12739   
Vietnam              214356                   47981               13167   
Yemen                 27994                   17918               53611   
Zambia                12809                   30065               28395   
Zimbabwe              18169                   32741               29802   

                   HIV/AIDS  Drug Use Disorders  ...  Diabetes Mellitus  \
Country/Territory                                ...                      
Afghanistan            4282                7094  ...              93207   
Albania                  57                 634  ...               4055   
Algeria                6101               10612  ...              89035   
American Samoa           15                   0  ...                970   
Andorra                  85                   0  ...                198   
...                     ...                 ...  ...                ...   
Venezuela             46090                1664  ...             175790   
Vietnam              148838               19959  ...             544222   
Yemen                  6276                3718  ...              30812   
Zambia              1175563                 933  ...              54098   
Zimbabwe            1836042                2271  ...              71175   

                   Chronic Kidney Disease  Poisonings  \
Country/Territory                            

In [105]:
# Qual será o país que mais teve mortes por cada causa de morte?
dfLeaders = dfCountry.idxmax(axis=0)
dfLeaders


Meningitis                                            India
Alzheimer's Disease and Other Dementias               China
Parkinson's Disease                                   China
Nutritional Deficiencies                              India
Malaria                                             Nigeria
Drowning                                              China
Interpersonal Violence                               Brazil
Maternal Disorders                                    India
HIV/AIDS                                       South Africa
Drug Use Disorders                            United States
Tuberculosis                                          India
Cardiovascular Diseases                               China
Lower Respiratory Infections                          India
Neonatal Disorders                                    India
Alcohol Use Disorders                                Russia
Self-harm                                             India
Exposure to Forces of Nature            

In [106]:
dfCountry.idxmax(axis=0).mode()[0]


'India'

Dado a população numerosa da índia, faz sentido que ela lidere em número de óbitos para a maior parte das causas. Será que o Brasil lidera em alguma causa? Vejamos:

In [107]:
# Em quais causas de morte o Brasil aparece no dataset dfLeaders?
dfLeaders[dfLeaders == "Brazil"]

Interpersonal Violence    Brazil
dtype: object

O Brasil é o país onde houve mais mortes por Violência Interpessoal nos anos de 1990 a 2019.

Vamos agora construir algumas visualizações.

### Visualização

Vamos começar analisando a causa de morte que o Brasil Lidera: a Interpesonal Violence.

In [108]:
import plotly.express as px

# Gráfico de linhas para mostrar a evolução do número de mortes por INTERPERSONAL VIOLENCE no Brasil

fig = px.line(df, x="Year", y="Interpersonal Violence", color="Country/Territory", title="Interpersonal Violence")
fig.show()

Faremos, agora, um gráfico de barras que mostra o número de mortes por cada causa no Brasil em 2019 - a data mais recente.

In [109]:
# Gráfico de barras com o número de mortes em cada país por cada causa em 2019

df2019 = df[df["Year"] == 2019]
df2019 = df2019.loc[df["Country/Territory"] == "Brazil"].drop(columns=["Country/Territory","Code", "Year"]).reset_index(drop=True)

# Agora usamos melt para transformar as colunas em linhas
df2019= df2019.melt(var_name="Cause", value_name="Deaths")
bar = px.bar(df2019,
            x="Deaths",
            y="Cause",
            color="Cause",
        color_discrete_sequence=["yellow", "blue", "green"],
)
# Inserir título
bar.update_layout(title="Deaths in Brazil in 2019")

# Não mostrar a legenda
bar.update_layout(showlegend=False)

bar



E no mundo? Como se distribuiram as causas em 2019?

In [110]:
# Gráfico de barras com o número total de mortes no planeta por cada causa de morte em 2019
bar2 = px.bar(dfFrequence.loc[2019],
            x=dfFrequence.loc[2019].values,
            y=dfFrequence.loc[2019].index,
            color=dfFrequence.loc[2019].index,
            color_discrete_sequence=px.colors.sequential.RdBu,
)
# Inserir título
bar2.update_layout(title="Deaths in the world in 2019")

# Não mostrar a legenda
bar2.update_layout(showlegend=False)

# Não mostrar títulos dos eixos
bar2.update_xaxes(title_text="")
bar2.update_yaxes(title_text="")

bar2





Essas foram algumas das centenas de visualizações interessantes que essa base de dados nos permite fazer.